src/net.h

```diff
@@ -609,8 +609,33 @@ class CNodeStats
 
 
 
-
+/** Transport protocol agnostic message container.
+ * Ideally it should only contain receive time, payload,
+ * command and size.
+ */
 class CNetMessage {
+public:
+    CDataStream m_recv;                  // received message data
+    int64_t m_time = 0;                  // time (in microseconds) of message receipt.
+    bool m_valid_netmagic = false;
+    bool m_valid_header = false;
+    bool m_valid_checksum = false;
+    uint32_t m_message_size = 0;         // size of the payload
+    std::string m_command;
+
+    CNetMessage(const CDataStream& recv_in) : m_recv(std::move(recv_in)) {}
+
+    void SetVersion(int nVersionIn)
```

Comment:
In commit "Refactor: split network transport deserializing from message container" (6294ecdb8bb4eb7049a18c721ee8cb4a53d80a06)

Why do CNetMessage and TransportDeserializer both have SetVersion methods now? It seems like only TransportDeserializer should have a SetVersion method, and it should be responsible for passing it the version on to CNetMessage objects as it constructs them.

Rght now both objects have SetVersion methods and the TransportDeserializer one is never called. I'd would suggest shifting more logic to TransportDeserializer, or at least deleting the SetVersion implementation that is never called.